### PR TITLE
[docs] Update gin.md

### PR DESCRIPTION
**Pull request by ymahajan**
- We need to take out this page altogether, we shouldn't need separate page for GIN indexes, this should be part of Indexes page. 
- Need to cleanup unnecessary stuff from this page.

### CODE REVIEW

File: docs/content/latest/explore/ysql-language-features/gin.md [hunk 1]
```diff
@@ -47,14 +47,12 @@ However, extension support is still in progress:
 
 ### CREATE INDEX
 
-Create the index using `USING ybgin` to specify the index access method:
+Create the index using `USING gin` to specify the index access method:
 
 ```sql
-CREATE INDEX ON mytable USING ybgin (mycol);
-```
+CREATE INDEX <optional name> ON <table> USING GIN (<column>);
```
Comment (jasonyb) [inline docs/content/latest/explore/ysql-language-features/gin.md:53]:
```suggestion
CREATE INDEX [name] ON <table> USING gin (<column>);
```

Please be consistent and say `USING gin`, not `USING GIN`.

File: docs/content/latest/explore/ysql-language-features/gin.md [hunk 2]
```diff
@@ -47,14 +47,12 @@ However, extension support is still in progress:
...
 
-The `gin` access method is reserved for temporary relations while `ybgin` is for Yugabyte-backed relations.
-You can still specify `USING gin`, and, if `mytable` is not a temporary table, it will be automatically substituted for `ybgin`.
```
Comment (jasonyb) [inline docs/content/latest/explore/ysql-language-features/gin.md:57]: This should still be called out in docs.  cc: @m-iancu

File: docs/content/latest/explore/ysql-language-features/gin.md [hunk 3]
```diff
@@ -347,11 +345,6 @@ SubDocKey(DocKey([], ["\x05one", EncodedSubDocKey(DocKey(0xfca0, [3], []), [])])
 SubDocKey(DocKey([], ["\x05thing", EncodedSubDocKey(DocKey(0x9eaf, [4], []), [])]), [SystemColumnId(0); HT{ physical: 1636678107997627 w: 1 }]) -> null; intent doc ht: HT{ physical: 1636678107961628 }
 ```
 
-## Roadmap
-
-GIN indexes are still in progress: see the GIN roadmap tracking [GitHub issue][github-roadmap] for more details.
```
Comment (polarweasel) [inline docs/content/latest/explore/ysql-language-features/gin.md:352]: Why are you taking this out? It seems important to let the reader know that this is an incomplete feature, and give them a link to the tracking ticket for it. We do this for other features.

File: docs/content/latest/explore/ysql-language-features/gin.md [hunk 4]
```diff
@@ -262,17 +260,17 @@ If write performance and storage aren't an issue, it may be worth creating a GIN
 
 ### Presplitting
 
-By default, `ybgin` indexes use a single range-partitioned tablet.
-Like regular tables and indexes, it is possible to presplit a `ybgin` index to multiple tablets at specified split points.
+By default, `gin` indexes use a single range-partitioned tablet.
+Like regular tables and indexes, it is possible to presplit a `gin` index to multiple tablets at specified split points.
```
Comment (jasonyb) [inline docs/content/latest/explore/ysql-language-features/gin.md:264]:
```suggestion
By default, GIN indexes use a single range-partitioned tablet.
Like regular tables and indexes, it is possible to presplit a GIN index to multiple tablets at specified split points.
```

This is incorrect when you look at "`gin`" as the access method name.  `gin` is for temporary tables, so it doesn't even use tablets.  It may be confusing to bring up the technically correct "`ybgin`" here, so I suggest saying the intentionally ambiguous "GIN".

File: docs/content/latest/explore/ysql-language-features/gin.md [hunk 5]
```diff
@@ -262,17 +260,17 @@ If write performance and storage aren't an issue, it may be worth creating a GIN
...
 These split points are for the index, so they need to be represented in the index key format.
 This is simple for tsvector and array types, but it gets complicated for jsonb and text (`pg_trgm`).
-`jsonb_path_ops` especially should use hash partitioning since the index key is itself a hash, but hash partitioning `ybgin` indexes is currently unsupported.
+`jsonb_path_ops` especially should use hash partitioning since the index key is itself a hash, but hash partitioning `gin` indexes is currently unsupported.
```
Comment (jasonyb) [inline docs/content/latest/explore/ysql-language-features/gin.md:267]:
```suggestion
`jsonb_path_ops` especially should use hash partitioning since the index key is itself a hash, but hash partitioning GIN indexes is currently unsupported.
```

File: docs/content/latest/explore/ysql-language-features/gin.md [hunk 6]
```diff
@@ -347,11 +345,6 @@ SubDocKey(DocKey([], ["\x05one", EncodedSubDocKey(DocKey(0xfca0, [3], []), [])])
 SubDocKey(DocKey([], ["\x05thing", EncodedSubDocKey(DocKey(0x9eaf, [4], []), [])]), [SystemColumnId(0); HT{ physical: 1636678107997627 w: 1 }]) -> null; intent doc ht: HT{ physical: 1636678107961628 }
 ```
 
-## Roadmap
-
-GIN indexes are still in progress: see the GIN roadmap tracking [GitHub issue][github-roadmap] for more details.
-
-[github-roadmap]: https://github.com/yugabyte/yugabyte-db/issues/7850
```
Comment (jasonyb) [inline docs/content/latest/explore/ysql-language-features/gin.md:354]: Why is this deleted?  Even if you have a legitimate reason, it's out of character with the rest of the changes that it should either be done in a separate PR or called out in the commit message (along with the ybgin to gin update).

**Review by jasonyb — CHANGES_REQUESTED** [review]
I did not check whether all "ybgin" in this file have been dealt with.

Make changes on other files if necessary: `docs/content/latest/api/ysql/the-sql-language/statements/ddl_create_index.md`, `docs/content/latest/explore/ysql-language-features/indexes-1.md`.

File: docs/content/latest/explore/ysql-language-features/gin.md (hunk 1 above)
Comment (jasonyb) [reply docs/content/latest/explore/ysql-language-features/gin.md:53]: Is `CREATE INDEX <optional name> ON <table> USING GIN (<column>);` really preferable over the simpler `CREATE INDEX ON mytable USING gin (mycol);`?  Especially optional grammar items like index name are unnecessary to mention here in my opinion.

File: docs/content/latest/explore/ysql-language-features/gin.md (hunk 3 above)
Comment (ymahajan) [reply docs/content/latest/explore/ysql-language-features/gin.md:352]: This is just Explore section. Also all this should go in Github Master ticket.

File: docs/content/latest/explore/ysql-language-features/gin.md (hunk 1 above)
Comment (ymahajan) [reply docs/content/latest/explore/ysql-language-features/gin.md:53]: Sure @jasonyb - i will take your suggestion and remove <optional>

File: docs/content/latest/explore/ysql-language-features/gin.md (hunk 2 above)
Comment (ymahajan) [reply docs/content/latest/explore/ysql-language-features/gin.md:57]: Not required for Explore section

File: docs/content/latest/explore/ysql-language-features/gin.md (hunk 6 above)
Comment (ymahajan) [reply docs/content/latest/explore/ysql-language-features/gin.md:354]: This is just Explore section. Also all this should go in Github Master ticket.

**Comment by polarweasel** [comment]
@ymahajan is this PR still required, now that #11374 is merged?
